build.gradle.kts

```diff
@@ -62,6 +63,19 @@ val downloadOpenapiSpec by tasks.register("downloadOpenapiSpec") {
         }
     }
 }
+val downloadObservabilityOpenapiSpec by tasks.register("downloadObservabilityOpenapiSpec") {
+    dependsOn(tasks.findByName("processTestResources"))
+    val observabilityApi = sourceSets.test.get().output.resourcesDir?.let { File("$it/observability-api.yml") }
+    onlyIf { observabilityApi?.exists()?.not() ?: false }
+    doLast {
+        observabilityApi?.let { fileSpec ->
+            fileSpec.parentFile.mkdirs()
+            download("https://eclipse-edc.github.io/Connector/openapi/observability-api/${libs.versions.edc.get()}/observability-api.yaml")
+                    .replace("example: null", "")
+                    .let { fileSpec.writeText(it) }
+        }
+    }
+}
```

Comment:
this is duplicated, please refactor in a way that the task registration can be parameterized and then call it twice (one for management and one for observability)